lib/views/welcome_view.dart

```diff
@@ -225,47 +250,277 @@ class _WelcomeViewState extends State<WelcomeView> with WidgetsBindingObserver {
                   builder: (BuildContext context) => PrivacyDialog(),
                 );
               },
-              style: TextButton.styleFrom(backgroundColor: Colors.blue),
+              style: TextButton.styleFrom(
+                  backgroundColor: Theme.of(context).colorScheme.primary),
               child: Text(
                 AppLocalizations.of(context)!.about_privacy,
-                style: TextStyle(color: Colors.white),
+                style:
+                    TextStyle(color: Theme.of(context).colorScheme.onPrimary),
               ),
             ),
           ],
         );
+      case "ask_permission_notification":
+        return Row(
```

Comment:
To stop this widget from growing way too big and hard to understand it would be nice to split every case into it's own widget rather than doing it all in this one.